aptos-move/aptos-gas/src/gas_meter.rs

```diff
@@ -445,9 +618,14 @@ impl GasMeter for AptosGasMeter {
         _ty: impl TypeView + 'a,
         args: impl ExactSizeIterator<Item = impl ValueView>,
     ) -> PartialVMResult<()> {
+        let num_args = NumArgs::new(args.len() as u64);
```

Comment:
Comment for `move_to` above: wouldn't we kind of release the memory? I know its still there in the cache, but conceptually, not longer part of the active memory reachable from the stack. Anyway, this can be also settled later.

Reply:
No for memory tracking purposes we must not consider it released. Otherwise someone could allocate 9.9 MB of data, move it into global storage and repeat.

Reply:
Yeah. I think the problem is perhaps that we should apply storage quota not on the write set, but when we actually execute the storage ops (like move_to or table::insert). Now a bit too late to change this, anyway.